src/main/scala/junctions/crossing.scala

```diff
@@ -91,3 +90,46 @@ object ClockToSignal {
     c2s.io.signal_out
   }
 }
+
+object LevelCrossing {
+  class SynchronizerBackend[T <: Data](typ: T, sync: Int, _clock: Clock) extends Module(Some(_clock)) {
+    val io = new Bundle {
+      val in = typ.asInput
+      val out = typ.asOutput
+    }
+
+    io.out := ShiftRegister(io.in, sync)
+  }
+
+  class SynchronizerFrontend[T <: Data](typ: T, _clock: Clock) extends Module(Some(_clock)) {
+    val io = new Bundle {
+      val in = typ.asInput
+      val out = typ.asOutput
+    }
+
+    io.out := RegNext(io.in)
+  }
+
+  def apply[T <: Data](from_clock: Clock, to_clock: Clock, in: T, sync: Int = 3): T = {
+    val front = Module(new SynchronizerFrontend(in, from_clock))
+    val back = Module(new SynchronizerBackend(in, sync - 1, to_clock))
+
+    front.io.in := in
+    back.io.in := front.io.out
+    back.io.out
+  }
+}
+
+object LevelTo {
```

Comment:
"LevelTo" is way too vague a name. "LevelSyncTo" makes more sense what it is actually doing.